charts/neon-broker/values.yaml

```diff
@@ -0,0 +1,124 @@
+# Default values for neon-broker.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+image:
+  # -- Neondatabase image repository
+  repository: neondatabase/neon
+  # -- image pull policy
+  pullPolicy: Always
+  # -- Overrides the image tag whose default is the chart appVersion.
+  tag: "latest"
+
+# -- Specify docker-registry secret names as an array
+imagePullSecrets: []
+
+# -- String to partially override neon-broker.fullname template (will maintain the release name)
+nameOverride: ""
+# -- String to fully override neon-broker.fullname template
+fullnameOverride: ""
+
+# Neon broker settings
+settings: {}
+
+serviceAccount:
+  # serviceAccount.create - Specifies whether a service account should be created
+  create: true
+  # serviceAccount.annotations -- Annotations to add to the service account
+  annotations: {}
+  # serviceAccount.name - The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+# -- Annotations for neon-broker pods
+podAnnotations: {}
+
+# -- Additional labels for neon-broker pods
+podLabels: {}
+  # neon_service: broker
+  # neon_env: staging
+  # neon_region: us-west-1
+  # neon_region_slug: virginia
+
+# -- neon-broker's pods Security Context
+podSecurityContext: {}
+  # fsGroup: 2000
+
+# -- neon-broker's containers Security Context
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
+  # service.annotations -- Annotations to add to the service
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: external  # use newer AWS Load Balancer Controller
+    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
+    service.beta.kubernetes.io/aws-load-balancer-scheme: internal  # deploy LB to private subnet
+    # assign service to this name at external-dns
+    external-dns.alpha.kubernetes.io/hostname: neon-broker.local
+  # service.type -- Service type
+  type: LoadBalancer
```

Comment:
I would prefer defaulting to kubernetes default here, which is `ClusterIP` and quite often just left out. The reasoning would come down to few points:

*  Ingress is more flexible in a way there can be multiple implementations (nginx, AWS, GCP)  even in the same cluster.
   * The `Service` remains the same regardless of whether `Ingress` even exists or not. Pretty common practice is to create ingress conditionally, and actually default to `false`.
* It should not be too incorrect to say `Ingress` tends to be the de facto nowdays
* `LoadBalancer` type expects one to have external load balancer, able to allocate ip addresses - one per service. This most likely isn't the case when trying to use the chart locally with kind/minikube and others - or even bare metal kubernetes for that matter. [metallb](https://metallb.universe.tf/)  exists, but isn't very popular option to my knowledge. 

So while it might seem Ingress adds complexity, from user's/operator's perspective it's less complicated, because there's no need to be able to allocate addresses. Exception is naturally services that don't operate over http.

Reply:
https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/server/ingress.yaml could perhaps serve as pretty decent example, while it may have more flexibility our use case needs.

Reply:
Well, won't then Ingress be more suitable for default?

Reply:
> Why should this be avoided? As I see it, that’s something rather widely used from CDNs to cloud providers.

Generally perhaps, but here we surely can spare a private IP, and not exposing allows to change routes without changing charts, and generally write less yml.

I'm not really against, but right now I'd like to merge this stuff, as it blocks main PR which replaces etcd usage throughout storage code, which requires more and more rebases. We can iterate later.

Reply:
I do feel you about writing a lot of boilerplate, that's unfortunately what kubernetes is - with generic helm charts a bit more.

To reduce the time spent, we should definitely come up with instructions to use `helm create my-helm-chart` as a starting point, because it will give decent defaults and also generates the most common objects, so often getting by with just `rm` editor.

> I'm not really against, but right now I'd like to merge this stuff, as it blocks main PR which replaces etcd usage throughout storage code, which requires more and more rebases. We can iterate later.

Would you be OK if I create the ingress here? Having done that a lot, doesn't require that much of my time and we would get this moving _while_ embracing some common practices. Frankly I would have even expected to just get a request to write the chart and would have done so without a question, so I'm really greatful you started working on this in the first place.

Reply:
> Would you be OK if I create the ingress here? 

Sure, go ahead :)

Reply:
Ok, lets get this moving :)